truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/impl/Accessor.java

```diff
@@ -416,6 +419,26 @@ protected String toString(TruffleLanguage<?> language, Env env, Object obj) {
         return API.toString(language, env, obj);
     }
 
+    protected boolean supportsOnLoopCount() {
+        return false;
+    }
+
+    protected void onLoopCount(Node source, int count) {
+        // optimized calltarget is not existent on default runtimes
+        if (OPTIMIZEDCALLTARGET != null) {
+            if (OPTIMIZEDCALLTARGET.supportsOnLoopCount()) {
+                OPTIMIZEDCALLTARGET.onLoopCount(source, count);
+            } else {
+                // needs an additional compatibilty check so older graal runtimes
+                // still run with newer truffle versions
+                RootNode root = source.getRootNode();
+                if (root instanceof LoopCountReceiver) {
+                    root.reportLoopCount(count);
```

Comment:
I think this should be `getRootNode().getCallTarget() instanceof LoopCountReceiver`.